### PR TITLE
Get rid of `previous-compilation-data.bin` file in META-INF

### PR DESCRIPTION
**Pull request by mvicsokolova**
`previous-compilation-data.bin` is Gradle's internal machinery file that was packed in the resulting artifact because we depended directly on the output of `compileJavaModuleInfo` task, depending on `compileJavaModuleInfo.destinationDirectory` solves the problem.

Similar problems:
https://github.com/Kotlin/kotlinx-datetime/issues/304
https://github.com/Kotlin/kotlinx.coroutines/issues/3668

### CODE REVIEW

File: integration-testing/build.gradle [hunk 1]
```diff
@@ -0,0 +1,89 @@
+
+buildscript {
```
Comment (ilya-g) [inline integration-testing/build.gradle:2]: How do you intend to run this project? Will you set up another build on TC?

**Review by ilya-g — APPROVED** [review]

File: integration-testing/build.gradle (hunk 1 above)
Comment (mvicsokolova) [reply integration-testing/build.gradle:2]: I'll add the second step in the build on TC that will depend on `publishToMavenLocal`

**Comment by Nek-12** [comment]
When will a 0.23.0 be published? This is causing https://github.com/Kotlin/kotlinx.coroutines/issues/3668 to occur despite using 1.7.3 as a coroutines dependency

**Comment by mvicsokolova** [comment]
Hi! I'm publishing 0.23.0 with this fix today

**Pull request merged**